### PR TITLE
[jenkins] fix: groovy linting changes

### PR DESCRIPTION
**Pull request by Wayonb**
## What is the current behavior?
Groovy linting is failing

## What's the issue?
New version of npm-groovy-lint was release which requires all warnings and info to be fix.

## How have you changed the behavior?
Fixed all Groovy lint warnings and info messages

## How was this change tested?
locally by running the lint script

### CODE REVIEW

File: jenkins/catapult/jenkins/catapult-client-build-catapult-project.groovy [hunk 1]
```diff
@@ -186,10 +199,9 @@ pipeline {
 				stage('run tests') {
 					steps {
 						script {
-							if (isCustomTestImage())
-								testImageName = "registry.hub.docker.com/symbolplatform/symbol-server-test:${buildImageLabel}"
-							else
-								testImageName = "symbolplatform/symbol-server-test:${buildImageLabel}"
+							testImageName = isCustomTestImage() ?
+									"registry.hub.docker.com/symbolplatform/symbol-server-test:${buildImageLabel}" :
+									"symbolplatform/symbol-server-test:${buildImageLabel}"
```
Comment (Jaguar0625) [inline jenkins/catapult/jenkins/catapult-client-build-catapult-project.groovy:204]: can you format so that ? and : are at the beginning of the (next) lines?

File: jenkins/catapult/jenkins/catapult-client-prepare-base-image.groovy [hunk 2]
```diff
@@ -46,10 +45,12 @@ pipeline {
 					version = properties[params.OPERATING_SYSTEM]
 
 					sanitizer = SANITIZER_BUILD.toBoolean() ? 'Sanitizer' : ''
-
-					dockerfileTemplate = "./jenkins/catapult/templates/${params.OPERATING_SYSTEM.capitalize()}${params.IMAGE_TYPE.capitalize()}${sanitizer}BaseImage.Dockerfile"
+					filename = "${params.OPERATING_SYSTEM.capitalize()}${params.IMAGE_TYPE.capitalize()}${sanitizer}"
+					dockerfileTemplate = "./jenkins/catapult/templates/${filename}BaseImage.Dockerfile"
 					dockerfileContents = readFile(file: dockerfileTemplate)
-					baseImage = 'windows' == "${OPERATING_SYSTEM}" ? "mcr.microsoft.com/windows/servercore:ltsc${version}" : "${params.OPERATING_SYSTEM}:${version}"
+					baseImage = 'windows' == "${OPERATING_SYSTEM}" ?
```
Comment (Jaguar0625) [inline jenkins/catapult/jenkins/catapult-client-prepare-base-image.groovy:51]: also here

File: jenkins/catapult/jenkins/catapult-client-release-build.groovy [hunk 3]
```diff
@@ -31,7 +31,7 @@ pipeline {
 	stages {
 		// stage('preliminary') {
 		//	 when {
-		//		 expression { is_public_build() && SHOULD_PUBLISH_BUILD_IMAGE.toBoolean() }
+		//		 expression { isPublicBuild() && SHOULD_PUBLISH_BUILD_IMAGE.toBoolean() }
 		//	 }
 
 		//	 steps {
```
Comment (Jaguar0625) [inline jenkins/catapult/jenkins/catapult-client-release-build.groovy:37]: should you delete this commented out code instead?

File: linters/groovy/.groovylintrc.json [hunk 4]
```diff
@@ -2,13 +2,19 @@
...
+
+    "dry.DuplicateNumberLiteral": { "enabled": false },
+    "dry.DuplicateStringLiteral": { "enabled": false },
+
     "formatting.LineLength": { "length": 140 },
     "formatting.Indentation": { "spacesPerIndentLevel": 1 },
     "formatting.NoTabCharacter": "off",
 
-    "size.ParameterCount": { "maxParameters": 9 },
+    "InvertedCondition": "off",
```
Comment (Jaguar0625) [inline linters/groovy/.groovylintrc.json:14]: how many warnings was this?

File: jenkins/shared-library/vars/monorepoSeedPipeline.groovy [hunk 5]
```diff
@@ -94,5 +94,5 @@ void call(Closure body) {
 
 String resolveOwnerAndProject(String gitUrl) {
 	String[] tokens = gitUrl.tokenize('/')
-	return "${tokens[2]}/${tokens.last().split('\\.')[0]}"
+	return [tokens[2], tokens.last().split('\\.')[0]]
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/monorepoSeedPipeline.groovy:97]: why this change?

File: linters/groovy/.groovylintrc.json [hunk 6]
```diff
@@ -2,13 +2,19 @@
   "extends": "recommended",
 
   "rules": {
+    "convention.CompileStatic": "off",
```
Comment (Jaguar0625) [inline linters/groovy/.groovylintrc.json:5]: what does this do?

File: jenkins/catapult/jenkins/catapult-client-release-build.groovy (hunk 3 above)
Comment (Wayonb) [reply jenkins/catapult/jenkins/catapult-client-release-build.groovy:37]: yea, will remove

File: linters/groovy/.groovylintrc.json (hunk 4 above)
Comment (Wayonb) [reply linters/groovy/.groovylintrc.json:14]: There are about 8 of them I think.
I disable it here since I thought this is the standard?

File: jenkins/shared-library/vars/monorepoSeedPipeline.groovy (hunk 5 above)
Comment (Wayonb) [reply jenkins/shared-library/vars/monorepoSeedPipeline.groovy:97]: the git url was getting parse twice, once in this method and another above.
With this change the git url only get parse once and the required two parts are returned to the caller.

File: linters/groovy/.groovylintrc.json (hunk 6 above)
Comment (Wayonb) [reply linters/groovy/.groovylintrc.json:5]: It disable this warning in the jenkinsfile

```
jenkins/catapult/jenkins/catapult-client-build-base-image-all.groovy
  1     warning  Class should be marked with one of @GrailsCompileStatic, @CompileStatic or @CompileDynamic  CompileStatic 
  ```

File: linters/groovy/.groovylintrc.json (hunk 4 above)
Comment (Jaguar0625) [reply linters/groovy/.groovylintrc.json:14]: i think we have this enabled for JS but check with @gimre-xymcity

**Review by Jaguar0625 — APPROVED** [review]
